### PR TITLE
llm: filter noisy llama-server logs

### LINKED ISSUE #16897: Very verbose llama-server logging by default since recent update
### What is the issue?

Hi, and thanks for the great work!

After updating to the latest version, journald gets flooded with per-request logs from the underlying `llama-server`, even DEBUG is disabled:

```
slot launch_slot_: id 4 | task -1 | sampler chain: logits -> penalties -> ...
slot update_slots: id 4 | task 80 | new prompt, n_ctx_slot = 28160, ...
slot print_timing: id 4 | task 80 | prompt eval time = ...
srv  update_slots: all slots are idle
```

Is it intended that this much gets logged by default? If there's a supported way to reduce it that I missed, I'd appreciate a pointer. Thanks!

### Relevant log output

```shell
Jun 25 10:38:02 ai4 ollama[1636975]: slot get_availabl: id  4 | task -1 | selected slot by LCP similarity, sim_best = 1.000 (> 0.100 thold), f_keep = 0.651
Jun 25 10:38:02 ai4 ollama[1636975]: slot launch_slot_: id  4 | task -1 | sampler chain: logits -> penalties -> ?dry -> ?top-n-sigma -> top-k -> ?typical -> ?top-p -> ?min-p -> ?xtc -> temp-ext -> dist
Jun 25 10:38:02 ai4 ollama[1636975]: slot launch_slot_: id  4 | task -1 | sampler params:
Jun 25 10:38:02 ai4 ollama[1636975]:         repeat_last_n = 64, repeat_penalty = 1.100, frequency_penalty = 0.000, presence_penalty = 0.000
Jun 25 10:38:02 ai4 ollama[1636975]:         dry_multiplier = 0.000, dry_base = 1.750, dry_allowed_length = 2, dry_penalty_last_n = 28160
Jun 25 10:38:02 ai4 ollama[1636975]:         top_k = 40, top_p = 1.000, min_p = 0.000, xtc_probability = 0.000, xtc_threshold = 0.100, typical_p = 1.000, top_n_sigma = -1.000, temp = 0.000
Jun 25 10:38:02 ai4 ollama[1636975]:         mirostat = 0, mirostat_lr = 0.100, mirostat_ent = 5.000, adaptive_target = -1.000, adaptive_decay = 0.900
Jun 25 10:38:02 ai4 ollama[1636975]: slot launch_slot_: id  4 | task 3794 | processing task, is_child = 0
Jun 25 10:38:02 ai4 ollama[1636975]: slot process_sing: id  0 | task -1 | saving idle slot to prompt cache
Jun 25 10:38:02 ai4 ollama[1636975]: slot process_sing: id  1 | task -1 | saving idle slot to prompt cache
Jun 25 10:38:02 ai4 ollama[1636975]: slot process_sing: id  2 | task -1 | saving idle slot to prompt cache
Jun 25 10:38:02 ai4 ollama[1636975]: slot process_sing: id  3 | task -1 | saving idle slot to prompt cache
Jun 25 10:38:02 ai4 ollama[1636975]: slot update_slots: id  4 | task 3794 | new prompt, n_ctx_slot = 28160, n_keep = 4, task.n_tokens = 69
Jun 25 10:38:02 ai4 ollama[1636975]: slot update_slots: id  4 | task 3794 | Checking checkpoint with [0, 64] against 0...
Jun 25 10:38:02 ai4 ollama[1636975]: slot update_slots: id  4 | task 3794 | restored context checkpoint (pos_min = 0, pos_max = 64, n_tokens = 65, n_past = 64, size = 1.524 MiB)
Jun 25 10:38:02 ai4 ollama[1636975]: slot update_slots: id  4 | task 3794 | cached n_tokens = 64, memory_seq_rm [64, end)
Jun 25 10:38:02 ai4 ollama[1636975]: slot update_slots: id  4 | task 3794 | cached n_tokens = 65, memory_seq_rm [65, end)
Jun 25 10:38:02 ai4 ollama[1636975]: slot init_sampler: id  4 | task 3794 | init sampler, took 0.04 ms, tokens: text = 69, total = 69
Jun 25 10:38:02 ai4 ollama[1636975]: slot print_timing: id  4 | task 3794 | prompt eval time =      23.60 ms /     5 tokens (    4.72 ms per token,   211.85 tokens per second)
Jun 25 10:38:02 ai4 ollama[1636975]: slot print_timing: id  4 | task 3794 |        eval time =     217.72 ms /    38 tokens (    5.73 ms per token,   174.53 tokens per second)
Jun 25 10:38:02 ai4 ollama[1636975]: slot print_timing: id  4 | task 3794 |       total time =     241.33 ms /    43 tokens
Jun 25 10:38:02 ai4 ollama[1636975]: slot print_timing: id  4 | task 3794 |    graphs reused =       3750
Jun 25 10:38:02 ai4 ollama[1636975]: slot      release: id  4 | task 3794 | stop processing: n_tokens = 106, truncated = 0
Jun 25 10:38:02 ai4 ollama[1636975]: srv  update_slots: all slots are idle
Jun 25 10:38:02 ai4 ollama[1636975]: [GIN] 2026/06/25 - 10:38:02 | 200 |   802.01326ms |    x.x.x.x | POST     "/v1/chat/completions"
```

### OS

_No response_

### GPU

_No response_

### CPU

_No response_

### Ollama version

0.30.10

### PR DESCRIPTION
**Pull request by samrusani**
Fixes #16897.

## Summary
- keep llama-server launched with `--log-verbosity 4` so memory and offload lines remain available for scheduler accounting
- continue parsing every runner log write, then filter default forwarding of noisy per-request `slot`, `srv`, sampler-continuation, and successful Gin access lines
- keep scheduler accounting lines, error lines, non-success Gin access lines, and full raw output when `OLLAMA_DEBUG` enables debug logging

## Validation
- `git diff --check`
- `/usr/bin/env GOPATH=/tmp/ollama-gopath GOMODCACHE=/tmp/ollama-gopath/pkg/mod GOCACHE=/tmp/ollama-gocache go test ./llm -run 'TestMemoryParsingWriter|TestAppendLlamaServerLogArgsKeepsSchedulerVerbosity' -count=1`
- `/usr/bin/env GOPATH=/tmp/ollama-gopath GOMODCACHE=/tmp/ollama-gopath/pkg/mod GOCACHE=/tmp/ollama-gocache go test ./llm -count=1`

## Notes
This intentionally avoids the approach from #16899 and #16910: it does not lower llama-server verbosity. The server still receives the verbosity-4 memory/offload output it needs, and only default log forwarding is filtered.

### CODE REVIEW

**Comment by samrusani** [comment]
Thanks for the review. Addressed in 2649692.

What changed:
- failed `srv log_server_r` 4xx/5xx lines now pass through the filter and are captured by `StatusWriter`
- `srv ... got exception:` lines are preserved as status errors
- successful/noisy `srv` access lines remain suppressed when debug logging is off

Validation:
- `gofmt -w llm/status.go llm/status_test.go llm/llama_server_test.go`
- `GOCACHE=/private/tmp/ollama-go-cache go test ./llm -run 'TestStatusWriterCapturesErrorLine|TestMemoryParsingWriterFiltersNoisyRequestLogs|TestAppendLlamaServerLogArgsKeepsSchedulerVerbosity' -count=1`
- `GOCACHE=/private/tmp/ollama-go-cache go test ./llm -count=1`
- `git diff --check`

@codex review

File: llm/llama_server.go [hunk 1]
```diff
@@ -2699,7 +2701,89 @@ func (w *memoryParsingWriter) Write(b []byte) (int, error) {
...
+func isLlamaServerSchedulerAccountingLog(line string) bool {
+	return deviceFreeRegex.MatchString(line) ||
+		bufferSizeRegex.MatchString(line) ||
+		offloadedLayersRegex.MatchString(line) ||
+		fitOverflowingLayersRegex.MatchString(line)
+}
+
+func isNoisyLlamaServerLogLine(line string) bool {
+	return strings.HasPrefix(line, "slot ") ||
+		strings.HasPrefix(line, "srv ") ||
```
[the remark this answers is not in the document]
Comment (samrusani) [reply llm/llama_server.go:2785]: Fixed in 2649692. `statusErrorLine` now recognizes failed `srv log_server_r` 4xx/5xx lines and `srv ... got exception:` lines, so the filter forwards them and `StatusWriter` captures them. Successful/noisy `srv` lines remain suppressed.

Validation: `go test ./llm -run 'TestStatusWriterCapturesErrorLine|TestMemoryParsingWriterFiltersNoisyRequestLogs|TestAppendLlamaServerLogArgsKeepsSchedulerVerbosity' -count=1`; `go test ./llm -count=1`; `git diff --check`.

File: llm/status.go [hunk 2]
```diff
@@ -181,9 +181,46 @@ func statusErrorLine(line string) string {
...
+	if strings.Contains(trimmed, "got exception:") {
+		return strings.TrimRight(trimmed, " \t\r")
+	}
+
+	if !strings.Contains(trimmed, "log_server_r:") {
+		return ""
+	}
+	for _, field := range strings.Fields(trimmed) {
+		if isFailedHTTPStatusField(field) {
+			return strings.TrimRight(trimmed, " \t\r")
```
[the remark this answers is not in the document]
Comment (samrusani) [reply llm/status.go:214]: Fixed in 5d3f2c4. `srv log_server_r` lines for `/health` are ignored before failed-status capture, so normal loading/no-slot health 503 probes stay suppressed and do not accumulate in `LastError`. Non-health 4xx/5xx `srv` lines and `srv ... got exception:` lines are still preserved.

Validation: `go test ./llm -run 'TestStatusWriterCapturesErrorLine|TestMemoryParsingWriterFiltersNoisyRequestLogs|TestAppendLlamaServerLogArgsKeepsSchedulerVerbosity' -count=1`; `go test ./llm -count=1`; `git diff --check`.

**Comment by samrusani** [comment]
Updated in 5d3f2c4 for the health-check follow-up.

What changed:
- `/health` `srv log_server_r` lines are ignored before failed-status capture, so normal loading/no-slot 503 probes stay suppressed and do not fill `LastError`
- non-health 4xx/5xx `srv` request lines and `srv ... got exception:` lines still pass through and are captured

Validation:
- `gofmt -w llm/status.go llm/status_test.go llm/llama_server_test.go`
- `GOCACHE=/private/tmp/ollama-go-cache go test ./llm -run 'TestStatusWriterCapturesErrorLine|TestMemoryParsingWriterFiltersNoisyRequestLogs|TestAppendLlamaServerLogArgsKeepsSchedulerVerbosity' -count=1`
- `GOCACHE=/private/tmp/ollama-go-cache go test ./llm -count=1`
- `git diff --check`

@codex review

**Comment by samrusani** [comment]
Withdrawing this PR to reduce stale contributor backlog. I can reopen it if maintainers want the work continued.
